### PR TITLE
mc: Inherit global flags for initialization messages.

### LINKED ISSUE #2089: mc: Global flags do not affect initialization messages
## Expected behaviour
Global flags `--no-color`, `--quiet`, `--json`, and `--debug` should be used for all output
All output should be de-colorized/supressed/json/verbose as specified
## Actual behaviour
The initialization messages are output with color in plaintext regardless of the specified options
```
$ mc --no-color --quiet --json --config-folder=/tmp config host list
mc: Configuration written to ‘/tmp/config.json’. Please update your access credentials.
mc: Successfully created ‘/tmp/share’.
mc: Initialized share uploads ‘/tmp/share/uploads.json’ file.
mc: Initialized share downloads ‘/tmp/share/downloads.json’ file.
{"status":"success","alias":"gcs  ","URL":"https://storage.googleapis.com","accessKey":"YOUR-ACCESS-KEY-HERE","secretKey":"YOUR-SECRET-KEY-HERE","api":"S3v2"}
```
image

## Steps to reproduce the behaviour
1. Run `mc` with one of the above options with a `config-folder` that has not been previously used

## mc version
```
Version: 2017-02-06T20:16:19Z
Release-tag: RELEASE.2017-02-06T20-16-19Z
Commit-id: 2c8115de4edc5612525488b1b3b804689d336d01
```
## System information
Bash shell on Ubuntu

### PR DESCRIPTION
**Pull request by harshavardhana**
Fixes #2089

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -174,14 +175,31 @@ func getSystemData() map[string]string {
 	}
 }
 
+type initMessage struct {
```
Comment (vadmeste) [inline cmd/main.go]: Can you add a comment here?

File: cmd/main.go [hunk 2]
```diff
@@ -174,14 +175,31 @@ func getSystemData() map[string]string {
...
+	Status  string `json:"status"`
+	Message string `json:"msg"`
+}
+
+func (m initMessage) String() string {
+	return m.Message
```
Comment (vadmeste) [inline cmd/main.go]: Missing some colors here (it was green)

**Comment by harshavardhana** [comment]
Fixed all the comments @vadmeste

**Comment by harshavardhana** [comment]
> With this output, since we don't have a type field it will be tough for the parsers to know if it is an init message or a config-list message. Better not to intermix messages of different types? If we want to intermix then the message should contain a type field for the parsers to differentiate.

Json is free from those problems if the relevant fields don't exist they just are nil. So it is not a big deal this is just for consistency. Ideally this case is never hit and it is usually never parsed in this manner.

**Comment by vadmeste** [comment]
I see it is better if we don't print init related messages when json is activated. It doesn't seem that someone will need that. Besides as @krishnasrinivas said, it intermixes json structure.

**Comment by harshavardhana** [comment]
> I see it is better if we don't print init related messages when json is activated. It doesn't seem that someone will need that. Besides as @krishnasrinivas said, it intermixes json structure.

Made the change check again @vadmeste @krishnasrinivas - please!

**Pull request merged**